tests/integration/helpers.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Helpers for the slurmd integration tests."""
+
+import logging
+import pathlib
+import shlex
+import subprocess
+import yaml
+
+from typing import Dict
+from urllib import request
+
+logger = logging.getLogger(__name__)
+
+ETCD = "etcd-v3.5.0-linux-amd64.tar.gz"
+ETCD_URL = f"https://github.com/etcd-io/etcd/releases/download/v3.5.0/{ETCD}"
+METADATA = yaml.safe_load(pathlib.Path("./metadata.yaml").read_text())
+NHC = METADATA["resources"]["nhc"]["filename"]
+NHC_URL = f"https://github.com/mej/nhc/releases/download/1.4.3/{NHC}"
```

Comment:
I know I am one to talk, but I think this is a little complicated :sweat_smile:. Why don't you hard code the version of the NHC tarball? I hardly imagine it will change underfoot while we work on the charm.

Reply:
I saw in other charms they were using this design and I like how it adds a bit of a dynamic nature to the tests where the resource is directly from the metadata.yaml. In this case though it doesn't add much benefit so I'll hardcode it.